test/AutoDiff/Sema/differentiable_attr_type_checking.swift

```diff
@@ -1158,6 +1146,42 @@ final class FinalClass: Differentiable {
   }
 }
 
+// Test `inout` parameters.
+
+@differentiable(wrt: y)
```

Comment:
Note: I intentionally didn't add tests for `@differentiable(jvp:vjp)`, since that's actively being removed (TF-1162).